packages/babel-parser/src/parser/base.js

```diff
@@ -16,21 +16,21 @@ export default class BaseParser {
 
   // Initialized by Tokenizer
   state: State;
-  input: string;
 
   isReservedWord(word: string): boolean {
     if (word === "await") {
       return this.inModule;
     } else {
-      return reservedWords[6](word);
+      return isES2015ReservedWord(word);
     }
   }
 
   hasPlugin(name: string): boolean {
-    return Object.hasOwnProperty.call(this.plugins, name);
+    return this.plugins.has(name);
   }
 
   getPluginOption(plugin: string, name: string) {
-    if (this.hasPlugin(plugin)) return this.plugins[plugin][name];
+    // $FlowIssue
+    if (this.hasPlugin(plugin)) return this.plugins.get(plugin)[name];
```

Comment:
Just out of curiosity, what does flow complain about here?

Reply:
It is because of this: https://github.com/facebook/flow/issues/2751 
Flow does not realize that calling `has` ensures that `get` will return something `!=null` 
Other way would be just calling get and testing if `!= null`.